pwnagotchi/plugins/default/wardriver.py

```diff
@@ -0,0 +1,248 @@
+            except Exception:
+                model = 'unknown'
+            # OS version
+            try:
+                with open('/etc/os-release', 'r') as release_info:
+                    release = release_info.read().split('\n')[0].split('=')[-1].replace('"', '')
+            except Exception:
+                release = 'unknown'
+            # Pwnagotchi name
+            device = data['main']['name']
+            # Pwnagotchi display model
+            display = data['ui']['display']['type'] # Pwnagotchi display
+            # CPU model
+            try:
+                with open('/proc/cpuinfo', 'r') as cpu_model:
+                    board = cpu_model.read().split('\n')[1].split(':')[1][1:]
+            except Exception:
+                board = 'unknown'
+            
+            # Brand: currently set equal to model
+            brand = model
+
+            return {
+                'file_format': file_format,
+                'app_release': app_release,
+                'model': model,
+                'release': release,
+                'device': device,
+                'display': display,
+                'board': board,
+                'brand': brand
+            }
+ 
+    def __new_wardriving_session(self):
+        self.ready = False
+        now = datetime.now()
+        self.__session_reported = []
+        session_name = now.strftime('%Y-%m-%dT%H:%M:%S')
+        session_file = os.path.join(self.__csv_path, f'{session_name}.csv')
+        logging.info(f'[WARDRIVER] Initializing new session file {session_file}')
+        self.__session_file = session_file
+        try:
+            with open(self.__session_file, 'w') as csv:
+                # See: https://api.wigle.net/csvFormat.html
+                # CSV pre-header
+                preheader_data = self.__wigle_info()
+                csv.write(f'{preheader_data["file_format"]},{preheader_data["app_release"]},{preheader_data["model"]},{preheader_data["release"]},{preheader_data["device"]},{preheader_data["display"]},{preheader_data["board"]},{preheader_data["brand"]}\n')
+                # CSV header
+                csv.write('MAC,SSID,AuthMode,FirstSeen,Channel,RSSI,CurrentLatitude,CurrentLongitude,AltitudeMeters,AccuracyMeters,Type\n')
+
+                logging.info('[WARDRIVER] Session file initialized. Ready to wardrive!')
+                self.ready = True
+        except Exception as e:
+            logging.critical(f'[WARDRIVER] Error while creating session file! {e}')
+
+    def __filter_whitelist_aps(self, unfiltered_aps):
+        '''
+        Filter whitelisted networks
+        '''
+        filtered_aps = [ ap for ap in unfiltered_aps if ap['hostname'] not in self.__whitelist ]
+        return filtered_aps
+    
+    def __filter_reported_aps(self, unfiltered_aps):
+        '''
+        Filter already reported networks
+        '''
+        filtered_aps = [ ap for ap in unfiltered_aps if (ap['mac'], ap['hostname']) not in self.__session_reported ]
```

Comment:
This should check against `main.whitelist`, instead of the plugin private whitelist.